Debugging information corrupt

https://github.com/dotnet/runtime/pull/66061#issuecomment-1057129704

Looks related to multimodule mode that is not shipping.

```
MultiModule.obj : warning LNK4078: multiple '.bss' sections found with different attributes (C0401040) [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.Reflection.Core.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.Reflection.Metadata.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.TypeLoader.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.Interop.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.Reflection.Execution.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
Framework.lib(System.Private.StackTraceMetadata.obj) : warning LNK4209: debugging information corrupt; recompile module; linking object as if no debug info [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
LINK : fatal error LNK1318: Unexpected PDB error; OK (0) '' [D:\a\_work\1\s\src\tests\nativeaot\SmokeTests\MultiModule\MultiModule.csproj] [D:\a\_work\1\s\src\tests\build.proj]
```

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
@@ -16,7 +16,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <!-- Set defaults for unspecified properties -->
   <PropertyGroup>
-    <NativeLib Condition="'$(OutputType)' == 'Library' and '$(NativeLib)' == ''">Shared</NativeLib>
+    <NativeLib Condition="'$(OutputType)' == 'Library' and '$(NativeLib)' == '' and '$(IlcMultiModule)' != 'true'">Shared</NativeLib>
     <NativeIntermediateOutputPath Condition="'$(NativeIntermediateOutputPath)' == ''">$(IntermediateOutputPath)native\</NativeIntermediateOutputPath>
     <NativeOutputPath Condition="'$(NativeOutputPath)' == ''">$(OutputPath)native\</NativeOutputPath>
     <NativeCompilationDuringPublish Condition="'$(NativeCompilationDuringPublish)' == ''">true</NativeCompilationDuringPublish>

--- a/src/tests/nativeaot/SmokeTests/MultiModule/MultiModule.csproj
+++ b/src/tests/nativeaot/SmokeTests/MultiModule/MultiModule.csproj
@@ -8,9 +8,6 @@
     <!-- This test always runs as multimodule. It's not supported if we don't have framework object files. -->
     <CLRTestTargetUnsupported Condition="'$(BuildNativeAotFrameworkObjects)' != 'true'">true</CLRTestTargetUnsupported>
     <IlcMultiModule>true</IlcMultiModule>
-
-    <!-- https://github.com/dotnet/runtime/issues/66191 -->
-    <CLRTestTargetUnsupported>true</CLRTestTargetUnsupported>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="MultiModule.cs" />

